chunkdot/chunkdot_sparse.py

```diff
@@ -179,6 +193,7 @@ def chunkdot_sparse(matrix_left, matrix_right, top_k, chunk_size, return_type="f
         right_n_cols,
         top_k,
         chunk_size,
+        progress_bar=progress_bar,
     )
     return csr_matrix(
```

Comment:
It hangs since the context manager is never closed.

```suggestion

    if progress_bar:
        progress_bar.close()
        
    return csr_matrix(
```